tidewave-core/src/acp_proxy.rs

```diff
@@ -595,19 +602,59 @@ pub async fn unit_testable_ws_handler<W, R>(
     state.websocket_senders.remove(&websocket_id);
     state.websocket_to_process.remove(&websocket_id);
 
-    // Remove session mappings for this WebSocket
-    let mut sessions_to_remove = Vec::new();
+    // Capture sessions for this WebSocket and remove mappings
+    let mut sessions_for_websocket = Vec::new();
     for entry in state.session_to_websocket.iter() {
         if *entry.value() == websocket_id {
-            sessions_to_remove.push(entry.key().clone());
+            sessions_for_websocket.push(entry.key().clone());
         }
     }
 
-    for session_id in sessions_to_remove {
-        state.session_to_websocket.remove(&session_id);
+    for session_id in &sessions_for_websocket {
+        state.session_to_websocket.remove(session_id);
     }
 
     debug!("WebSocket connection closed: {}", websocket_id);
+
+    // Spawn a task to send session/cancel after 10 seconds
+    let state_clone = state.clone();
```

Comment:
There is a race condition in this code. If you refresh the page and then, exactly 10 seconds later, you refresh the page again, you may cancel in this brief interview.

There are different ways to solve this, the simplest is probably by having an increasing integer that you bump whenever someone connect. Then you read it here and you only cancel if the integer is the same by the time you disconnect.